sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/RsaKeyCryptographyClient.java

```diff
@@ -0,0 +1,223 @@
+        } catch (InvalidKeyException | NoSuchAlgorithmException | NoSuchPaddingException | IllegalBlockSizeException | BadPaddingException e) {
+            return Mono.error(e);
+        }
+    }
+
+    Mono<byte[]> unwrapKeyAsync(KeyWrapAlgorithm algorithm, byte[] encryptedKey, Context context, JsonWebKey jsonWebKey) {
+
+        keyPair = getKeyPair(jsonWebKey);
+
+        // Interpret the requested algorithm
+        Algorithm baseAlgorithm = AlgorithmResolver.Default.get(algorithm.toString());
+
+        if (baseAlgorithm == null) {
+            if(serviceCryptoAvailable()) {
+                return serviceClient.unwrapKeyAsync(algorithm, encryptedKey, context);
+            }
+            return Mono.error(new NoSuchAlgorithmException(algorithm.toString()));
+        } else if (!(baseAlgorithm instanceof AsymmetricEncryptionAlgorithm)) {
+            return Mono.error(new NoSuchAlgorithmException(algorithm.toString()));
+        }
+
+        if (keyPair.getPrivate() == null){
+            if(serviceCryptoAvailable()) {
+                return serviceClient.unwrapKeyAsync(algorithm, encryptedKey, context);
+            }
+            return Mono.error(new IllegalArgumentException("Private portion of the key not available to perform unwrap operation"));
+        }
+
+        AsymmetricEncryptionAlgorithm algo = (AsymmetricEncryptionAlgorithm) baseAlgorithm;
+
+        ICryptoTransform transform;
+
+        try {
+            transform = algo.CreateDecryptor(keyPair);
+            return Mono.just(transform.doFinal(encryptedKey));
+        } catch (InvalidKeyException | NoSuchAlgorithmException | NoSuchPaddingException | IllegalBlockSizeException | BadPaddingException e) {
+            return Mono.error(e);
+        }
+    }
+
+
+    Mono<SignResult> signDataAsync(SignatureAlgorithm algorithm, byte[] data, Context context, JsonWebKey key) {
+        try {
+            HashAlgorithm hashAlgorithm = SignatureHashResolver.Default.get(algorithm);
+            MessageDigest md = MessageDigest.getInstance(hashAlgorithm.toString());
+            md.update(data);
+            byte[] digest = md.digest();
+            return signAsync(algorithm, digest, context, key);
+        } catch (NoSuchAlgorithmException e){
+            return Mono.error(e);
+        }
+    }
+
+
+    Mono<Boolean> verifyDataAsync(SignatureAlgorithm algorithm, byte[] data, byte[] signature, Context context, JsonWebKey key) {
+        HashAlgorithm hashAlgorithm = SignatureHashResolver.Default.get(algorithm);
+        try {
+            MessageDigest md = MessageDigest.getInstance(hashAlgorithm.toString());
+            md.update(data);
+            byte[] digest = md.digest();
+            return verifyAsync(algorithm, digest, signature, context, key);
+        } catch (NoSuchAlgorithmException e){
+            return Mono.error(e);
+        }
+    }
+
+    private boolean serviceCryptoAvailable(){
+        return serviceClient != null ;
+    }
```

Comment:
What is the scenario where serviceClient is null?  Shouldn't service crypto always be available, since these instances are created by the CryptographyClient?

Reply:
If someone creates a key locally and passes it as JsonWebKey, i.e key doesn't exist in KV,  for example an Oct Key.
We wouldn't have any key Id associated with it. will not have info about the vault url to create service client / send requests to.

Should we enforce user to always specify vault endpoint in the Cryptography client builder?    Currently I am parsing vault url from the key identifier.

Reply:
No, the builder should just take a JWK or a Key ID like you have now.  It's possible that you could enforce that the JWK has valid kid field but I think what you have here is good for now.